pytype/pytd/typeshed.py

```diff
@@ -144,32 +144,41 @@ def _load_third_party_packages(self):
     stubs/ contains type information for third-party packages. Each top-level
     directory corresponds to one PyPI package and contains one or more modules,
     plus a metadata file (METADATA.toml). If there are separate Python 2 stubs,
-    they live in an @python2 subdirectory. METADATA.toml
-    takes @python2 into account, so if a package has both foo.pyi and
-    @python2/foo.pyi, METADATA.toml will contain `python2 = True`.
+    they live in an @python2 subdirectory. If stubs support both Python 2 and
+    Python 3 without a separate @python2 directory, METADATA.toml will contain
+    `python2 = True`.
 
     Returns:
       A mapping from module name to a set of
       (package name, major_python_version) tuples.
     """
     metadata = {}
     modules = collections.defaultdict(set)
+    py2 = False
+    py3 = False
     for third_party_file in self._list_files("stubs"):
       parts = third_party_file.split(os.path.sep)
       if parts[-1] == "METADATA.toml":  # {package}/METADATA.toml
         _, metadata_file = self._load_file(
             os.path.join("stubs", third_party_file))
         metadata[parts[0]] = toml.loads(metadata_file)
-      elif "@python2" not in parts:  # {package}/{module}
+      elif "@python2" in parts:  # {package}/@python2/{module}
+        py2 = True
+      elif "@python2" not in parts: # {package}/{module}
+        if parts[-1].endswith(".pyi"):
```

Comment:
Why is this check needed?

Reply:
There can be other files in the top-level directory, for example the `@tests` directory or a `README` file. The existence of these files shouldn't mean that the stubs support Python 3.

Reply:
Makes sense, thanks!